zerver/tests/tests.py

```diff
@@ -1615,6 +1623,30 @@ def test_illegal_name_changes(self):
                                        dict(full_name='x' * 1000))
         self.assert_json_error(json_result, 'Name too long!')
 
+    def test_illegal_characters_in_name_changes(self):
+        # type: () -> None
+        email = 'hamlet@zulip.com'
+        self.login(email)
+        user = get_user_profile_by_email(email)
+        full_name = user.full_name
+
+        with self.settings(NAME_CHANGES_DISABLED=True):
```

Comment:
Ok, I don't think you need to duplicate the things that are already tested in the other test.